src/restapi/mm_routes.ts

```diff
@@ -0,0 +1,49 @@
+import fs from 'fs';
+
+import express, {Request, Response} from 'express';
+import {AppCall} from 'mattermost-redux/types/apps';
+
+import app from '../app/app';
+
+import {jsonStoreFileName} from '../app/constants';
+
+const router = express.Router();
+
+router.post('/createform', async (req: Request, res: Response) => {
+    const appCall: AppCall = req.body;
+    if (appCall.type === 'form') {
+        const createForm = app.calls.createForm(appCall.context.post.message);
+        res.json(createForm);
+    } else {
+        const message = app.createTicketFromPost(appCall);
+        res.json({});
+    }
+});
+
+router.get('/manifest.json', (req: Request, res: Response) => {
+    res.json(app.manifest);
```

Comment:
Oh I see the app's fields are being accessed like this, outside of the `App` class. That's a little too implicit imo, as there's no way to know that by looking at the `App` class. Can we have the `App` implement a `getManifest` function to be more explicit?

Reply:
I agree here. Do you think I should also implement method that are used for getting the manifest fields, or is it ok to access them directly?  getManifest().root_url?

Reply:
Since this function wants the whole manifest, it needs to have access to the manifest itself. For other fields accessed on the manifest, I think helper methods on `App` would be good like `app.getRootURL()`